rest/data/distro.go

```diff
@@ -74,3 +78,49 @@ func DeleteDistroById(distroId string) error {
 	}
 	return nil
 }
+
+type CopyDistroOpts struct {
+	DistroIdToCopy string
+	NewDistroId    string
+}
+
+func CopyDistro(ctx context.Context, env evergreen.Environment, u *user.DBUser, opts CopyDistroOpts) error {
+	distroToCopy, err := distro.FindOneId(opts.DistroIdToCopy)
+	if err != nil {
+		return errors.Wrapf(err, "finding distro '%s'", opts.DistroIdToCopy)
+	}
+	if distroToCopy == nil {
+		return gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    fmt.Sprintf("distro '%s' not found", opts.DistroIdToCopy),
+		}
+	}
+
+	distroToCopy.Id = opts.NewDistroId
```

Comment:
This will be caught by the validator, but it may be preferable if we check that `opts.NewDistroId` is different from `distroToCopy.Id` explicitly, which would let us pass back a 400 error instead of 500 in this case to indicate that it's a user input error.